libs/sdk-backend-spi/src/workspace/dashboards/layout/utils.ts

```diff
@@ -0,0 +1,235 @@
+     * it replaces them in the layout provided to FluidLayoutTransforms instance instead)
+     */
+    public filterRows = (pred: (params: { row: TRow; rowIndex: number }) => boolean): this => {
+        this._layout.rows = this._layout.rows.filter((row, rowIndex) => pred({ row, rowIndex }));
+        return this;
+    };
+
+    /**
+     * Update layout rows.
+     * Callback is called for each row, and row is replaced by the returned value.
+     * (Similar to Array.map, but it does not return new rows array,
+     * it replaces them in the layout provided to FluidLayoutTransforms instance instead)
+     */
+    public updateRows = (callback: (params: { row: TRow; rowIndex: number }) => TRow): this => {
+        this._layout.rows = this._layout.rows.map((row, rowIndex) => callback({ row, rowIndex }));
+        return this;
+    };
+
+    /**
+     * Reduce layout rows.
+     * (Similar to Array.reduce)
+     */
+    public reduceRows = <TResult>(
+        callback: (acc: TResult, params: { row: TRow; rowIndex: number }) => TResult,
+        initialValue: TResult,
+    ): TResult => {
+        return this._layout.rows.reduce((rowsAcc, row, rowIndex) => {
+            return callback(rowsAcc, { row, rowIndex });
+        }, initialValue);
+    };
+
+    /**
+     * Update layout columns.
+     * Callback is called for each column, and column is replaced by the returned value.
+     * (Similar to Array.map, but it does not return new columns array,
+     * it replaces them in the layout provided to FluidLayoutTransforms instance instead)
+     */
+    public updateColumns = (
+        callback: (params: { column: TColumn; row: TRow; columnIndex: number; rowIndex: number }) => TColumn,
+    ): this => {
+        this.updateRows(({ row, rowIndex }) => {
+            row.columns = row.columns.map((column, columnIndex) =>
+                callback({ row, rowIndex, column, columnIndex }),
+            );
+            return row;
+        });
+
+        return this;
+    };
+
+    /**
+     * Reduce layout columns.
+     * (Similar to Array.reduce)
+     */
+    public reduceColumns = <TResult>(
+        callback: (
+            acc: TResult,
+            params: { column: TColumn; row: TRow; columnIndex: number; rowIndex: number },
+        ) => TResult,
+        initialValue: TResult,
+    ): TResult => {
+        return this.reduceRows((rowsAcc, { row, rowIndex }) => {
+            return row.columns.reduce((columnsAcc, column, columnIndex) => {
+                return callback(columnsAcc, { row, rowIndex, column, columnIndex });
+            }, rowsAcc);
+        }, initialValue);
+    };
+
+    /**
+     * Returns transformed layout
+     */
+    public layout(): IFluidLayout<TContent, TColumn, TRow> {
+        return this._layout;
+    }
+}
+
+/**
+ * Has fluid layout only empty rows and columns?
+ * @alpha
+ */
+export const isFluidLayoutEmpty = (layout: IFluidLayout<any>): boolean => {
```

Comment:
Should this be generic? In case we need to use the type inside in the future? So that the API will not have to change for it.